include/listmovementcontrol.h

```diff
@@ -0,0 +1,207 @@
+#ifndef NEWSBOAT_LISTMOVEMENTCONTROL_H_
+#define NEWSBOAT_LISTMOVEMENTCONTROL_H_
+
+#include "listwidgetbackend.h"
+#include <string>
+
+namespace newsboat {
+
+// Backend is required to implement:
+// std::uint32_t get_height()
+// std::uint32_t get_num_lines()
+// void update_position(std::uint32_t pos, std::uint32_t scroll_offset)
+// virtual void on_list_changed()
+
+template<class Backend>
+class ListMovementControl : public Backend {
+public:
+	using Backend::Backend;
+
+	// Make sure the backend's destructor is marked virtual
+	~ListMovementControl() override = default;
+
+	void set_num_context_lines(std::uint32_t context_lines)
+	{
+		num_context_lines = context_lines;
+
+		// Force a recalculation of the scroll offset
+		set_position(current_position);
+	}
```

Comment:
This method is always called after the constructor, and is never called anywhere else. I suggest replacing it with a constructor:

```c++
ListMovementControl::ListMovementControl(const std::string& list_name, Stfl::Form& form,
	std::uint32_t scrolloff)
    : Backend(list_name, form)
{
    num_context_lines = scrollof;
    set_position(current_position);
}
```

`Backend`'s constructor can be made `protected` to avoid people calling it by accident.

Reply:
Makes sense, done